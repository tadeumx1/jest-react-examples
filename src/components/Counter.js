import React, { Component } from 'react';

class Counter extends Component {

  render() {
    return (
      <div>
          <h2> VocÃª tem { this.props.count * 2 } counters </h2>
      </div>
    );
  }
}

export default Counter;