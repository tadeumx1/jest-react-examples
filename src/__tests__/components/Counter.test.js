import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Counter from '../../components/Counter';

describe('Testing Counter Component', () => {

  it('should render correctly', () => {

    const wrapper = shallow(

      <Counter count={5} />

    );

    expect(wrapper).toMatchSnapshot();

    // Mudar o count de 5 para 10

    // Dessa forma ele vai criar dois Snapshots desse componente
    // um quando o valor do count era 5 e outro quando era 10

    wrapper.setProps({ count: 10 });

    expect(wrapper).toMatchSnapshot();

  });

});
