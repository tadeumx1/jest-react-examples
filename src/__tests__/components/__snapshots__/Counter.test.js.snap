// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    count={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
         Você tem 
        10
         counters 
</h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " Você tem ",
          10,
          " counters ",
        ],
      },
      "ref": null,
      "rendered": Array [
        " Você tem ",
        10,
        " counters ",
      ],
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
           Você tem 
          10
           counters 
</h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Você tem ",
            10,
            " counters ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Você tem ",
          10,
          " counters ",
        ],
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Testing Counter Component should render correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    count={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
         Você tem 
        20
         counters 
</h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " Você tem ",
          20,
          " counters ",
        ],
      },
      "ref": null,
      "rendered": Array [
        " Você tem ",
        20,
        " counters ",
      ],
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
           Você tem 
          20
           counters 
</h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Você tem ",
            20,
            " counters ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Você tem ",
          20,
          " counters ",
        ],
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
